native/src/widget/grid.rs

```diff
@@ -0,0 +1,276 @@
+//! Write some text for your users to read.
+use crate::{
+    layout::{
+        flex::{self, Axis},
+        Limits, Node,
+    },
+    Align, Element, Hasher, Layout, Length, Point, Size, Widget,
+};
+use std::{any::TypeId, hash::Hash, iter};
+
+/// A container that distributes its contents in a grid.
+///
+/// # Example
+///
+/// ```
+/// # use iced_native::{renderer::Null, Element, Grid as NativeGrid, Text};
+/// # type Grid<'a> = NativeGrid<'a, (), Null>;
+/// Grid::with_children(
+///     vec![
+///         Element::from(Text::new("First row, first column")),
+///         Element::from(Text::new("First row, second column")),
+///         Element::from(Text::new("Second row, first column")),
+///         Element::from(Text::new("Second row, second column")),
+///     ],
+/// )
+/// .columns(2);
+/// ```
+#[allow(missing_debug_implementations)]
+pub struct Grid<'a, Message, Renderer> {
+    columns: Option<usize>,
+    column_width: Option<u16>,
+    elements: Vec<Element<'a, Message, Renderer>>,
+}
+
+impl<'a, Message, Renderer> Grid<'a, Message, Renderer> {
+    /// Create a new empty [`Grid`].
+    ///
+    /// [`Grid`]: struct.Grid.html
+    pub fn new() -> Self {
+        Self::with_children(Vec::new())
+    }
+
+    /// Create a new [`Grid`] with the given [`Element`]s.
+    ///
+    /// [`Grid`]: struct.Grid.html
+    /// [`Element`]: ../struct.Element.html
+    pub fn with_children(
+        elements: Vec<Element<'a, Message, Renderer>>,
+    ) -> Self {
+        Self {
+            columns: None,
+            column_width: None,
+            elements,
+        }
+    }
+
+    /// Sets a fixed amount of columns for the [`Grid`].
+    ///
+    /// [`Grid`]: struct.Grid.html
+    pub fn columns(mut self, columns: usize) -> Self {
```

Comment:
1.  Should I make a note in the documentation about how 0 is handled?
2.  I could also return an error when a 0 is passed.
3.  We could also make the type here NonZeroUsize.
4.  Is `usize` fine?

Reply:
If `0` is passed, I would skip layout and draw nothing.

Reply:
My assumption here is if we use `with_columns` and `with_column_width` we won't need this function anymore anyway.

Reply:
I handled this case in `fn layout` now.